types/state.go

```diff
@@ -97,7 +97,9 @@ func NewFromGenesisDoc(genDoc *types.GenesisDoc) (State, error) {
 		ConsensusParams:                  *genDoc.ConsensusParams,
 		LastHeightConsensusParamsChanged: genDoc.InitialHeight,
 	}
-	copy(s.AppHash[:], genDoc.AppHash)
+	appHash := make(Hash, 32)
```

Comment:
is this necessary? can you not just copy directly? `s.AppHash = genDoc.AppHash.Bytes()`

Reply:
You can yes, thanks for the suggestions, just added it.